src/Symfony/Component/Mime/SMimeMessage.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mime;
+
+/**
+ * @author Sebastiaan Stok <s.stok@rollerscapes.net>
+ *
+ * @experimental in 4.3
+ */
+class SMimeMessage extends RawMessage
+{
+    private $originalMessage;
+
+    /**
+     * @param iterable|string $message
+     */
+    public function __construct(RawMessage $originalMessage, $message)
+    {
+        parent::__construct($message);
+
+        $this->originalMessage = $originalMessage;
+    }
+
+    public function getOriginalMessage(): RawMessage
```

Comment:
What about marking this method as being `@internal` as we don't want people to change the original method **after** having signed/encrypted it.

Reply:
Also, as we don't care about the body here, just the headers, what about storing only the original headers? And we know that we will get an instance of `Headers` as we are not supporting `RawMessage` instances.